esp_repl/include/esp_repl.h

```diff
@@ -0,0 +1,156 @@
+/*
+ * SPDX-FileCopyrightText: 2025 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#pragma once
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdbool.h>
+#include "esp_err.h"
+#include "esp_linenoise.h"
+#include "esp_commands.h"
+
+/**
+ * @brief Handle to a REPL instance.
+ */
+typedef struct esp_repl_instance *esp_repl_handle_t;
+
+/**
+ * @brief Function prototype called before executing a command.
+ *
+ * @param ctx User-defined context pointer.
+ * @param buf Buffer containing the command.
+ * @param reader_ret_val Return value from the reader function.
+ *
+ * @return ESP_OK to continue execution, error code to abort.
+ */
+typedef esp_err_t (*esp_repl_pre_executor_fn)(void *ctx, char *buf, const esp_err_t reader_ret_val);
```

Comment:
The `const` on `esp_err_t` here isn't necessary since it will be passed by copy anyway (`int`).
However, a `const` on `buf` seems necessary to me to prevent having the executor modify the command, except if that's is a feature/expected behavior.